inc/Engine/Optimization/LazyRenderContent/Frontend/Processor/Dom.php

```diff
@@ -107,6 +146,16 @@ private function add_hash_to_element( $element, $depth, $html ) {
 			$child_html       = $child->ownerDocument->saveHTML( $child ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
 			$opening_tag_html = strstr( $child_html, '>', true ) . '>';
 
+			$exclusions_pattern = $this->get_exclusions_pattern();
+
+			if (
+				! empty( $exclusions_pattern )
+				&&
+				preg_match( '/(' . $exclusions_pattern . ')/i', $opening_tag_html )
```

Comment:
What about using RegexTrait find method?

https://github.com/wp-media/wp-rocket/blob/73fae2b3e7d2c7054f751bb9168cde175673d101/inc/Engine/Optimization/RegexTrait.php#L28-L45

Reply:
this method is intended to find elements we're going to interact with, not just check if a pattern is in a string, I think that would be overkill to use it in this context.